src/lib/Microsoft.Health.Fhir.Ingest/Service/NormalizationEventConsumerService.cs

```diff
@@ -104,19 +119,59 @@ private async Task<IContentTemplate> GetNormalizationTemplate()
 
         private async Task ConsumeAsyncImpl(IEnumerable<IEventMessage> events)
         {
-            var template = await GetNormalizationTemplate();
-
+            // Step 1: Get the normalization mapping template
+            IContentTemplate template = null;
+            try
+            {
+                template = await GetNormalizationTemplate();
+            }
+            catch (Exception ex)
+            {
+                ex.AddEventContext(events);

```

Comment:
currently the exception.Data does not appear in the logs, but for PaaS will still probably add some guardrails to remove the exception.Data in case the current behavior changes.